### PR TITLE
add appreceiver and install link handling

### PR DESCRIPTION
**Pull request by omercs**
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-android/pull/372%23discussion_r29386430%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-android/pull/372%23discussion_r29390380%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-android/pull/372%23discussion_r29479977%22%5D%2C%20%22comments%22%3A%20%7B%22Pull%20254c600900c22a5c8a6755c776295c34609e4a78%20src/src/com/microsoft/aad/adal/ApplicationReceiver.java%2058%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-android/pull/372%23discussion_r29386430%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22what%20happens%20when%20the%20if%20statement%20is%20false%3F%22%2C%20%22created_at%22%3A%20%222015-04-29T22%3A07%3A42Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/5713915%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kpanwar%22%7D%7D%2C%20%7B%22body%22%3A%20%22Application%20will%20receive%20intents%20with%20different%20actions.%20This%20code%20needs%20to%20consider%20only%20package%20added%20action.%20Other%20actions%20are%20not%20related%20here.%22%2C%20%22created_at%22%3A%20%222015-04-29T23%3A03%3A52Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/466805%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/omercs%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%22%2C%20%22created_at%22%3A%20%222015-04-30T23%3A25%3A59Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/5713915%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kpanwar%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20src/src/com/microsoft/aad/adal/ApplicationReceiver.java%3AL1-142%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [x] <a href='#crh-comment-Pull 254c600900c22a5c8a6755c776295c34609e4a78 src/src/com/microsoft/aad/adal/ApplicationReceiver.java 58'></a> &nbsp;<b><a href='https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/372#discussion_r29386430'>File: src/src/com/microsoft/aad/adal/ApplicationReceiver.java:L1-142</a></b>
- <a href='https://github.com/kpanwar'></a> what happens when the if statement is false?
- <a href='https://github.com/omercs'></a> Application will receive intents with different actions. This code needs to consider only package added action. Other actions are not related here.

<a href='https://www.codereviewhub.com/AzureAD/azure-activedirectory-library-for-android/pull/372?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/AzureAD/azure-activedirectory-library-for-android/pull/372?approve=1'></a>&nbsp;<a href='https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/372'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by RandalliLama** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/372'>Refresh</a>.<a href='#crh-update-none'></a>

File: src/src/com/microsoft/aad/adal/ApplicationReceiver.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+	public static final String INSTALL_URL_KEY = "app_link";
+
+	/**
+	 * This method receives message for any application status based on filters
+	 * defined in your manifest.
+	 */
+	@Override
+	public void onReceive(Context context, Intent intent) {
+		// Check if the application is install and belongs to the broker package
+		if (intent.getAction().equals(Intent.ACTION_PACKAGE_ADDED)) {
```
Comment (kpanwar) [inline src/src/com/microsoft/aad/adal/ApplicationReceiver.java]: what happens when the if statement is false?

Comment (omercs) [inline src/src/com/microsoft/aad/adal/ApplicationReceiver.java]: Application will receive intents with different actions. This code needs to consider only package added action. Other actions are not related here.

Comment (kpanwar) [inline src/src/com/microsoft/aad/adal/ApplicationReceiver.java]: :+1:

**Pull request merged**